tests/src/test/scala/com/nvidia/spark/rapids/timezone/TimeZonePerfSuite.scala

```diff
@@ -0,0 +1,172 @@
+    "2000-01-02 03:04:05Iran")
+
+  private val regularStrings = Array(
+    "1970-01-01 00:00:00",
+    "2000-01-02 00:00:00",
+    "2030-01-02 00:00:00")
+
+  /**
+   * create a data frame with schema:
+   * - c_ts: timestamp column
+   * - c_long_of_ts: long value which is microseconds
+   * - c_date: date column
+   * - c_int_of_date:int value which is days from 1970-01-01
+   * - c_str_for_cast: strings for cast to timestamp, formats are yyyy, yyyy-mm, ...
+   * - c_str_of_ts: strings with format: yyyy-MM-dd HH:mm:ss
+   */
+  def createDF(spark: SparkSession): DataFrame = {
+    val id = col("id")
+    val tsArray = Array[Long](year1980, year2000, year2030)
+    val dateArray = Array[Int](0, 100, 200)
+    val columns = Array[Column](
+      TimeZonePerfUtils.createColumn(id, TimestampType, TsGenFunc(tsArray)).alias("c_ts"),
+      TimeZonePerfUtils.createColumn(id, LongType, TsGenFunc(tsArray)).alias("c_long_of_ts"),
+      TimeZonePerfUtils.createColumn(id, DateType, DateGenFunc(dateArray)).alias("c_date"),
+      TimeZonePerfUtils.createColumn(id, IntegerType, DateGenFunc(dateArray)).alias("c_int_of_date"),
+      TimeZonePerfUtils.createColumn(id, StringType, StringGenFunc(stringsForCast)).alias("c_str_for_cast"),
+      TimeZonePerfUtils.createColumn(id, StringType, StringGenFunc(regularStrings)).alias("c_str_of_ts")
+    )
+    val range = spark.range(numRows)
+    range.select(columns: _*)
+  }
+
+  /**
+   * Run 6 rounds for both Cpu and Gpu,
+   * but only print the elapsed times for the last 5 rounds.
+   */
+  def runAndRecordTime(
+      testName: String,
+      func: (SparkSession, String) => DataFrame,
+      conf: SparkConf = new SparkConf()): Any = {
+
+    if (!enablePerfTest) {
+      // by default skip perf test
+      return None
+    }
+    println(s"test,type,zone,used MS")
+    for (zoneStr <- zones) {
+      // run 6 rounds, but ignore the 1 round.
+      for (i <- 1 to 6) {
+        // run on Cpu
+        val startOnCpu = System.nanoTime()
+        withCpuSparkSession(
+          spark => func(spark, zoneStr).collect(),
+          conf)
+        val endOnCpu = System.nanoTime()
+        val elapseOnCpuMS = (endOnCpu - startOnCpu) / 1000000L
+        if (i != 1) {
+          println(s"$testName,Cpu,$zoneStr,$elapseOnCpuMS")
+        }
+
+        // run on Gpu
+        val startOnGpu = System.nanoTime()
+        withGpuSparkSession(
+          spark => func(spark, zoneStr).collect(),
+          conf)
+        val endOnGpu = System.nanoTime()
+        val elapseOnGpuMS = (endOnGpu - startOnGpu) / 1000000L
+        if (i != 1) {
+          println(s"$testName,Gpu,$zoneStr,$elapseOnGpuMS")
+        }
+      }
+    }
+  }
+
+  test("test from_utc_timestamp") {
+    // cache time zone DB in advance
+    GpuTimeZoneDB.cacheDatabase()
+    Thread.sleep(5L)
+
+    def perfTest(spark: SparkSession, zone: String): DataFrame = {
+      spark.read.parquet(path).select(functions.max( // use max to reduce the result data
```

Comment:
`spark.read.parquet(path).cache().select(...)` Perhaps add cache here to eliminate the overhead of the Scan ?

Reply:
Because the test file generated is small 2.5M, so the second time running the OS will cache the file.
I tried to generate a file that is about 500M and use df.cache.
The previous result (2.5M) is:
```
CPU
from_utc_timestamp,Cpu,Iran,733
from_utc_timestamp,Cpu,Iran,781

GPU
from_utc_timestamp,Gpu,Iran,131
from_utc_timestamp,Gpu,Iran,114
```

The new result(500M + df.cache) is:
```
CPU:
from_utc_timestamp,Cpu,Iran,670
from_utc_timestamp,Cpu,Iran,665

GPU:
from_utc_timestamp,Gpu,Iran,334
from_utc_timestamp,Gpu,Iran,326
```
The event log shows GPU decode time is 115 ms.

So using highly compressed file can reduce the scan/decode time.